blog/tech-talk-window-resize-behavior.md

```diff
@@ -0,0 +1,290 @@
+
+---
+
+Recently, I worked on improving Electron and Chromium's window resize behavior.
+
+## The bug
+
+We were seeing an issue on Windows where old frames would become visible while resizing a window:
+
+![Animated GIF showing the issue where old frames would be shown while resizing windows](/assets/img/blog/tech-talk-window-resize-behavior/window-resize-bug-showing-old-frames.gif)
+
+## What made this bug particularly interesting?
+
+1. It was challenging.
+2. As you'll see later, there were two different bugs under the hood.
+
+## Fixing the bug
+
+With a bug like this, the first challenge is figuring out where to start looking.
+
+Electron builds upon Chromium. Chromium is the open source version of Google Chrome. When compiling Electron, Electron's source code is added into the Chromium source tree as a subdirectory. Electron then relies on Chromium's code to provide most of the functionality of a modern browser.
+
+Chromium has about 36 million lines of code. Electron is a large project, too. That is a lot of code that could be causing this issue.
+
+<!--truncate-->
+
+## Narrowing down the root cause
+
+I did a lot of experimentation.
+
+First, I noticed that the issue occurred in Google Chrome, too:
+
+![Screenshot of Google Chrome also showing the resize issue](/assets/img/blog/tech-talk-window-resize-behavior/google-chrome-repro.png)
+
+This suggested that the issue was likely in Chromium, not in Electron.
+
+Additionally, the issue was not visible on macOS. That suggested that it was in Windows-specific source code.
+
+## The crucial lead
+
+I tried a lot of different command line flags and configuration options.
+
+I noticed that [`app.disableHardwareAcceleration()`](../docs/latest/api/app#appdisablehardwareacceleration) fixed the issue. Without hardware acceleration, the issue was gone.
+
+Here is some context: Chromium supports various different graphics APIs for showing pixels on screen (OpenGL, Vulkan, Metal, and more). On Windows, it uses different graphics APIs than on Windows or Linux. Even on Windows, Chromium can work with multiple different graphics backends.
+
+Which graphics backend Chromium uses depends on the user's hardware. For example, some graphics backends require the computer to have a GPU.
+
+I tried various graphics backends and noticed that the following flags fixed the issue:
+
+- `--use-angle=warp`
+- `--use-angle=vulkan`
+- `--use-gl=desktop`
+- `--use-gl=egl`
+- `--use-gl=osmesa`
+- `--use-gl=swiftshader`
+
+The following flags reproduced the issue:
+
+- `--use-angle=d3d11` (this is currently the default on Windows)
+- `--use-angle=gl` (falls back to Direct3D 11 on Windows, see `chrome://gpu/`)
+
+None of the working flags were good enough to be used as the default in Electron apps on Windows. They were either too slow or lacked good driver support.
```

Comment:
```suggestion
None of the working flags were suitable as the default backend for Electron apps on Windows. They were either too slow or lacked good driver support.
```

Small edit to avoid repeating `good` in consecutive sentences

Reply:
Fixed in a slightly different way. Thanks!